src/plugins/dev_tools/public/application.tsx

```diff
@@ -64,6 +71,11 @@ function DevToolsWrapper({
     breadcrumbService.setBreadcrumbs(activeDevTool.title);
   }, [activeDevTool, docTitleService, breadcrumbService]);
 
+  useExecutionContext(appServices.executionContext, {
+    type: 'application',
+    page: activeDevTool.id,
+  });
```

Comment:
Looking at the definition of the properties in the Execution context in `src/core/types/execution_context.ts`, I would expect this to be:

```ts
{
  type: 'dev_tools',
  name: 'Dev Tools',
  id: activeDevTool.id,
  description: `Dev Tools > ${activeDevTool.title}`,
  url: window.location.href,
}
```

This is what I meant with my comment https://github.com/elastic/kibana/pull/124996/files#r808273840: Having all fields as optional leads to developers using them indistinctively. Then, making it harder to analyze and troubleshoot in the future.